### PR TITLE
BUG#100589 Custom field Constraints mashup doesn't work with Team States

### PR DESCRIPTION
**Pull request by pavel-drobushevich**
From #189

### CODE REVIEW

**Comment by pavel-drobushevich** [comment]
@roadhump Bug was tested by Natasha, plz merge

File: Custom Field Constraints/CFConstraints.state.interrupter.slice.js [hunk 1]
```diff
@@ -19,28 +19,68 @@ tau.mashups
...
+                    }.bind(this));
+                }.bind(this));
+            },
+
+            isProperState: function(stateName, state, workflowIds) {
+                return _.contains(workflowIds, state.workflow.id) && this.isStateEqual(stateName, state)
+            },
+
+            isStateEqual: function(expectedStateName, actualState) {
+                return (expectedStateName.toLowerCase() === actualState.name.toLowerCase()
```
Comment (victor-homyakov) [inline Custom Field Constraints/CFConstraints.state.interrupter.slice.js]: Maybe store `expectedStateName.toLowerCase()` to local variable?

File: Custom Field Constraints/CFConstraints.state.interrupter.slice.js [hunk 2]
```diff
@@ -19,28 +19,68 @@ tau.mashups
...
             },
 
-            _getEntityState: function(entity, entityStates, changes, defaultProcess) {
-                var stateName = this.sliceDecoder.decode(_.find(changes,function(change) {
+            _getEntityState: function(entity, entityStates, changes, defaultProcess, teamProjects) {
+                var change = _.find(changes, function(change) {
                     return this._shouldChangeBeHandled(change);
-                }, this).value);
+                }, this);
+                var stateName = this.sliceDecoder.decode(change.value);
```
Comment (victor-homyakov) [inline Custom Field Constraints/CFConstraints.state.interrupter.slice.js]: If `_.find()` won't find any change that should be handled, `change.value` throws.

**Comment by pavel-drobushevich** [comment]
@victor-homyakov Is it ok now?

File: Custom Field Constraints/CFConstraints.state.interrupter.slice.js [hunk 3]
```diff
@@ -19,28 +19,72 @@ tau.mashups
...
+                } else {
+                    return _.find(entityStates, function(state) {
+                        return state.process.id === this.dataProvider.getEntityProcessId(entity, defaultProcess)
+                            && state.entityType.name === entity.entityType.name
+                            && this.isStateEqual(stateName, state)
+                    }, this);
+                }
+            },
 
+            _getTeamState: function(stateName, entity, entityStates, change, teamProjects) {
```
Comment (victor-homyakov) [inline Custom Field Constraints/CFConstraints.state.interrupter.slice.js]: Argument `change` is not used in `_getTeamState: function(stateName, entity, entityStates, change, teamProjects)`

**Pull request merged**